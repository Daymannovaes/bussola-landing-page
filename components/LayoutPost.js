import React from 'react';
import PropTypes from 'prop-types';
import Head from 'next/head';

import Disqus from './Disqus';

import {
    colors, PADDING_UNIT, POST_SECTION_PADDING, POST_WIDTH, POST_ASSET_WIDTH,
} from '../config/style-guide';

const LayoutPost = ({
    children,
    title,
    date,
    id,
}) => (
    <section>
        <Head>
            <title>{title} - Bussola.school</title>
        </Head>

        <style jsx>
            {`
            :global(main) {
                background: ${colors.lightwhite} !important;
            }

            section {
                color: ${colors.lightblack};
                text-align: left;
                line-height: 1.6;
                letter-spacing: -.004em;

                margin: 0 auto;
                padding-left: ${POST_SECTION_PADDING}px
                padding-right: ${POST_SECTION_PADDING}px;

                font-size: 0.9em;
            }

            article :global(> *) {
                max-width: ${POST_WIDTH}px;
                margin-left: auto;
                margin-right: auto;
            }

            section :global(ul.LayoutPost-clear) {
                list-style: none;
            }

            section :global(p) {
                margin-top: ${6 * PADDING_UNIT}px;
                margin-bottom: 0;
            }

            section article h1 {
                font-size: 1.5em;
            }

            section :global(h1), section :global(h2), section :global(h3), section :global(h4) {
                text-shadow: initial;
                text-transform: initial;
                max-width: ${POST_ASSET_WIDTH}px !important;
                font-weight: 900;
                margin-top: ${6 * PADDING_UNIT}px;
            }

            section :global(h3), section :global(h4) {
                padding: ${2 * PADDING_UNIT}px ${4 * PADDING_UNIT}px;
                border-bottom: 4px dotted #00000020;
                transition: 200ms all;
            }

            section acticle :global(h3), section article :global(h4) {
                margin-bottom: -${3 * PADDING_UNIT}px;
            }

            section :global(h3:hover), section :global(h4:hover) {
                border-bottom: 4px dotted #00000050;
            }

            date {
                font-size: 0.9em;
                color: ${colors.lightblue};
            }
        `}
        </style>

        <article>
            <h1>{title}</h1>

            <date>{(new Date(date)).toDateString()}</date>

            {children}
        </article>

        <Disqus id={id} title={title} />
    </section>
);

LayoutPost.propTypes = {
    children: PropTypes.node.isRequired,
    title: PropTypes.string.isRequired,
    date: PropTypes.number.isRequired,
    id: PropTypes.string.isRequired,
};

export default LayoutPost;
