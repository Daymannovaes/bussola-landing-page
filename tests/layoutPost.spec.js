import React from 'react';
import LayoutPost from '../components/LayoutPost';
import renderer from 'react-test-renderer';

describe('layoutPost.spec.js', () => {
    it('snapchot', () => {
        const component = renderer.create(
            <LayoutPost title="Layout Title" id="layout-title" date={1539901476547}>This is a text</LayoutPost>
        );

        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

});
