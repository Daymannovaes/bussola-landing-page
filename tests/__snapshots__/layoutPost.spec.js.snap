// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`layoutPost.spec.js snapchot 1`] = `
<section
  className="jsx-228109686"
>
  <article
    className="jsx-228109686"
  >
    <h2
      className="jsx-228109686"
    >
      Layout Title
    </h2>
    <date
      className="jsx-228109686"
    >
      Thu Oct 18 2018
    </date>
    This is a text
  </article>
  <div>
    <h4>
      Coment√°rios
    </h4>
    <div
      id="disqus_thread"
    />
  </div>
</section>
`;
